specialization_week_1/specialization_week_1.py

```diff
@@ -0,0 +1,115 @@
+"""
+TASK 1
+Write a class to represent a Cash Register.
+You class should keep the state of price total and purchased items
+Below is a starter code:
+------------------------
+1. you can add new variables and function if you want to
+2. you will NEED to add accepted method parameters where required.
+For example, method add_item probably accepts some kind of an item?..
+3. You will need to write some examples of how your code works.
+"""
+
+class CashRegister:
+
+    def __init__(self):
+        self.total_items = {}
+
+    def add_item(self, item, price):
+        self.item = item
+        self.price = price
+        self.total_items[item] = price
+
+    def remove_item(self, item):
+        self.item = item
+        self.total_items.pop(item)
+
+    def apply_discount(self, discount): # discount must be entered as percentage
+        self.discount = discount
+        for name, price in self.total_items.items():
+            self.total_items[name] = self.total_items[name] - round((price * discount)/100, 2)
+
+
+    def get_total(self):
+        total_cost = sum([price for price in self.total_items.values()])
+        return total_cost
+
+    def show_items(self):
+        return self.total_items
+
+    def reset_register(self):
+        self.total_items = {}
+
+
+# EXAMPLE code run:
+basket = CashRegister()
+basket.add_item('fish', 10)
+basket.add_item('bread', 1.50)
+basket.add_item('chocolate', 1)
+basket.show_items()
+basket.get_total()
+basket.apply_discount(50)
+basket.show_items()
+basket.get_total()
+basket.remove_item('bread')
+basket.show_items()
+
+
+
+"""
+TASK 2
+Write a base class to represent a student. Below is a starter code. 
+Feel free to add any more new features to your class. 
+As a minimum a student has a name and age and a unique ID.
+Create a new subclass from student to represent a concrete student doing a specialization, for example:
+Software Student and Data Science student. 
+"""
+
+
+class Student:
```

Comment:
✅ Well done on Task 2
Point 8/10

-1 No specific attribute added to the child class `CFGStudent` i.e. `specialisation`
-0.5 Uncessary attributes `self.subject_name` and `self.subject_grade`
-0.5 `add_subject_grade` should be separated into two methods i.e. `add_subject` and `update_grade`. SRP principle.